src/Components/Wizard/StepMarker/types.ts

```diff
@@ -0,0 +1,29 @@
+import { ReactNode } from "react"
+
+export interface RenderProps {
+  nextStep: () => void
+  previousStep: () => void
+  gotoStep: (index: number) => void
+  stepState: State
+  isComplete: () => boolean
+}
+
+export interface Props {
+  children?: (renderProps: RenderProps) => ReactNode | void
+  steps: Step[]
+  style?: any
+  onChange?: any
+  currentStepIndex: number
+}
+
+export interface State {
+  currentStepIndex: number
+  steps: StepState[]
+}
+
+export type StepState = Step & { isActive: boolean; isComplete: boolean }
+
+export interface Step {
+  label: string
+  onClick?: any
+}
```

Comment:
I tried but didn't ultimately succeed yet at making a StepMarker, Generic Step and Form Step that would share/extend from each other. `WizardSchema`s are actually a `Step[]` so it would be nice to pass them around similarly to interested components.